Library/Homebrew/description_cache_store.rb

```diff
@@ -89,3 +90,49 @@ def select(&block)
     database.select(&block)
   end
 end
+
+#
+# {CaskDescriptionCacheStore} provides methods to fetch and mutate cask descriptions used
+# by the `brew desc` and `brew search` commands.
+#
+class CaskDescriptionCacheStore < DescriptionCacheStore
+  # If the database is empty `update!` it with all known casks.
+  #
+  # @return [nil]
+  def populate_if_empty!
+    return unless database.empty?
+
+    # TODO: 3.6.0: consider if we want to actually read all contents of all casks or odeprecate.
+    Cask::Cask.all.each { |c| update!(c.full_name, [c.name.join(", "), c.desc.presence]) }
+  end
+
+  # Use an update report to update the {CaskDescriptionCacheStore}.
+  #
+  # @param report [Report] an update report generated by cmd/update.rb
+  # @return [nil]
+  def update_from_report!(report)
+    return populate_if_empty! if database.empty?
+    return if report.empty?
+
+    alterations = report.select_formula(:AC) +
+                  report.select_formula(:MC)
+
+    update_from_cask_tokens!(alterations)
+    delete_from_cask_tokens!(report.select_formula(:DC))
+  end
+
+  # Use an array of cask tokens to update the {CaskDescriptionCacheStore}.
+  #
+  # @param cask_tokens [Array] the casks to update
+  # @return [nil]
+  def update_from_cask_tokens!(cask_tokens)
+    return populate_if_empty! if database.empty?
+
+    cask_tokens.each do |token|
+      c = Cask::CaskLoader.load(token)
+      update!(c.full_name, [c.name.join(", "), c.desc]) if c.present?
```

Comment:
This is intended to be hidden from non-cask OSes, but isn't.

Reply:
Turns out Linux is having trouble when it encounters `homebrew/cask-drivers/brother-p-touch-update-software`:
```
vagrant@ubuntu2004:/home/linuxbrew/.linuxbrew/Homebrew$ brew info --cask brother-p-touch-update-software --json=v2
Error: Cask 'brother-p-touch-update-software' is unreadable: undefined method `to_sym' for #<Version::NULL>
Did you mean?  to_s
               to_str
```
```
vagrant@ubuntu2004:/home/linuxbrew/.linuxbrew/Homebrew$ brew tap homebrew/cask-drivers
==> Tapping homebrew/cask-drivers
Cloning into '/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-cask-drivers'...
remote: Enumerating objects: 12646, done.
remote: Counting objects: 100% (1702/1702), done.
remote: Compressing objects: 100% (683/683), done.
remote: Total 12646 (delta 1144), reused 1556 (delta 1019), pack-reused 10944
Receiving objects: 100% (12646/12646), 2.43 MiB | 3.51 MiB/s, done.
Resolving deltas: 100% (8642/8642), done.
Tapped 207 casks (236 files, 3.1MB).
...
Error: undefined method `full_name' for nil:NilClass
Do not report this issue until you've run `brew update` and tried again.
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/description_cache_store.rb:136:in `rescue in block in update_from_cask_tokens!'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/description_cache_store.rb:132:in `block in update_from_cask_tokens!'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/description_cache_store.rb:131:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/description_cache_store.rb:131:in `update_from_cask_tokens!'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/tap.rb:335:in `block in install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cache_store.rb:26:in `use'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/tap.rb:333:in `install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cmd/tap.rb:65:in `tap'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:110:in `<main>'
```